webapp/components/Modal/DeleteUserModal.vue

```diff
@@ -0,0 +1,182 @@
+
+    <template slot="footer">
+      <base-button class="cancel" @click="cancel">{{ $t('actions.cancel') }}</base-button>
+      <base-button danger filled class="confirm" icon="exclamation-circle" @click="openModal">
+        {{ $t('settings.deleteUserAccount.name') }}
+      </base-button>
+    </template>
+  </ds-modal>
+</template>
+
+<script>
+import gql from 'graphql-tag'
+import { mapMutations } from 'vuex'
+import { SweetalertIcon } from 'vue-sweetalert-icons'
+import RelativeDateTime from '~/components/RelativeDateTime'
+import UserTeaser from '~/components/UserTeaser/UserTeaser'
+
+export default {
+  name: 'DeleteUserModal',
+  components: {
+    UserTeaser,
+    SweetalertIcon,
+    RelativeDateTime,
+  },
+  props: {
+    userdata: { type: Object, required: true },
+  },
+  data() {
+    return {
+      isOpen: true,
+      success: false,
+      loading: false,
+      // isAdmin: this.$store.getters['auth/isAdmin'],
+      isAdmin: true,
+    }
+  },
+  computed: {
+    title() {
+      return this.$t('settings.deleteUserAccount.name')
+    },
+
+    modalData(userdata) {
+      return function (userdata) {
+        return {
+          name: 'confirm',
+          data: {
+            type: userdata.name,
+            resource: userdata,
+            modalData: {
+              titleIdent: this.$t('settings.deleteUserAccount.accountWarningIsAdmin'),
+              messageIdent: this.$t('settings.deleteUserAccount.infoAdmin'),
+              messageParams: {},
+              buttons: {
+                confirm: {
+                  danger: true,
+                  icon: 'trash',
+                  textIdent: this.$t('settings.deleteUserAccount.confirmDeleting'),
+                  callback: () => {
+                    this.confirm(userdata)
+                  },
+                },
+                cancel: {
+                  icon: 'close',
+                  textIdent: this.$t('actions.cancel'),
+                  callback: () => {},
+                },
+              },
+            },
+          },
+        }
+      }
+    },
+  },
+  methods: {
+    ...mapMutations({
+      commitModalData: 'modal/SET_OPEN',
+    }),
+    openModal() {
+      this.commitModalData(this.modalData(this.userdata))
+    },
+    cancel() {
+      // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
+      // await this.modalData.buttons.cancel.callback()
+      this.isOpen = false
+      setTimeout(() => {
+        this.$emit('close')
+      }, 1000)
+    },
+    async confirm() {
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation($id: ID!, $resource: [Deletable]) {
+              DeleteUser(id: $id, resource: $resource) {
+                id
+              }
+            }
+          `,
+          variables: { id: this.userdata.id, resource: ['Post', 'Comment'] },
+        })
+        .then(({ _data }) => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-24T20:28:06Z" title="Tuesday, March 24th 2020, 9:28:06 pm +01:00">Mar 24, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,175 @@
+<template>
+  <ds-modal :title="title" :is-open="isOpen" @cancel="cancel">
+    <transition name="ds-transition-fade">
+      <ds-flex v-if="success" class="hc-modal-success" centered>
+        <sweetalert-icon icon="success" />
+      </ds-flex>
+    </transition>
+    <div>
+      <ds-section>
+        <ds-heading>{{ userdata.name }}</ds-heading>
+        <ds-placeholder>
+          <div>
+            <ds-avatar :name="userdata.name" :image="userdata.avatar" size="x-large" />
+          </div>
+          <br />
+          <ds-text>
+            Slug:
+            <ds-text size="x-large">{{ userdata.slug }}</ds-text>
+          </ds-text>
+          <ds-text>
+            Id:
+            <ds-text size="x-large">{{ userdata.id }}</ds-text>
+          </ds-text>
+          <ds-text>
+            contributionsCount:
+            <ds-text size="x-large">{{ userdata.contributionsCount }}</ds-text>
+          </ds-text>
+          <ds-text>
+            commentedCount:
+            <ds-text size="x-large">{{ userdata.commentedCount }}</ds-text>
+          </ds-text>
+          <ds-text>
+            createdAt:
+            <ds-text size="x-large">{{ userdata.createdAt }}</ds-text>
+          </ds-text>
+        </ds-placeholder>
+      </ds-section>
+    </div>
+
+    <template slot="footer">
+      <base-button class="cancel" @click="cancel">{{ $t('actions.cancel') }}</base-button>
+      <base-button danger filled class="confirm" icon="exclamation-circle" @click="openModal">
+        {{ $t('settings.deleteUserAccount.name') }}
+      </base-button>
+    </template>
+  </ds-modal>
+</template>
+
+<script>
+import gql from 'graphql-tag'
+import { mapMutations } from 'vuex'
+import { SweetalertIcon } from 'vue-sweetalert-icons'
+
+export default {
+  name: 'DeleteUserModal',
+  components: {
+    SweetalertIcon,
+  },
+  props: {
+    userdata: { type: Object, required: true },
+  },
+  data() {
+    return {
+      isOpen: true,
+      success: false,
+      loading: false,
+      isAdmin: this.$store.getters['auth/isAdmin'],
+    }
+  },
+  computed: {
+    title() {
+      return this.$t('settings.deleteUserAccount.name')
+    },
+
+    modalData(userdata) {
+      return function(userdata) {
+        return {
+          name: 'confirm',
+          data: {
+            type: userdata.name,
+            resource: userdata,
+            modalData: {
+              titleIdent: this.$t('settings.deleteUserAccount.accountWarningIsAdmin'),
+              messageIdent: this.$t('settings.deleteUserAccount.infoAdmin'),
+              messageParams: {},
+              buttons: {
+                confirm: {
+                  danger: true,
+                  icon: 'trash',
+                  textIdent: this.$t('settings.deleteUserAccount.confirmDeleting'),
+                  callback: () => {
+                    this.confirm(userdata)
+                  },
+                },
+                cancel: {
+                  icon: 'close',
+                  textIdent: this.$t('actions.cancel'),
+                  callback: () => {},
+                },
+              },
+            },
+          },
+        }
+      }
+    },
+  },
+  methods: {
+    ...mapMutations({
+      commitModalData: 'modal/SET_OPEN',
+    }),
+    openModal() {
+      this.commitModalData(this.modalData(this.userdata))
+    },
+    cancel() {
+      // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
+      // await this.modalData.buttons.cancel.callback()
+      this.isOpen = false
+      setTimeout(() => {
+        this.$emit('close')
+      }, 1000)
+    },
+    async confirm() {
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation($id: ID!, $resource: [Deletable]) {
+              DeleteUser(id: $id, resource: $resource) {
+                id
+              }
+            }
+          `,
+          variables: { id: this.userdata.id, resource: ['Post', 'Comment'] },
+        })
+        .then(({ _data }) => {
```


Prefer `async/await` over `then/catch`

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-05-28T13:44:08Z" title="Thursday, May 28th 2020, 3:44:08 pm +02:00">May 28, 2020</time>_

---


Not strongly, but I do the same.
See example here: `webapp/components/Modal/DisableModal.vue`

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-08-26T15:07:31Z" title="Wednesday, August 26th 2020, 5:07:31 pm +02:00">Aug 26, 2020</time>_

---


Is not fixed yet.
Is this intensionally?

You like it this way better?